doc: Harmonize linux sdk install docs + add udev rules guide to other linux distro guide

aaillet opened the pull request:
While following the "install linux host dependencies" guide to install Zephyr and it's dependencies on a fresh Fedora 41 install, when I first tried flashing a firmware on a board, I realized that the part of the guide concerning the installation of the udev rule was missing.
I think that it could be a bit harsh for a new user to find out what's the problem.

As a regular Zephyr user, I'm not using the `west sdk install` command but it could also be a way to rework this doc to use it instead (if it include the udev rule installation commands) if the solution i proposed is not following the current documentation spirit.

Harmonize the sdk install part of the "install linux host dependencies" with the common sdk installation guide under develop/toolchains/zephyr_sdk.html.
- Add udev rules installation guide
- Relocate some notes

aaillet committed:
doc: centralizes doc for installing SDK for Linux

- rename ubuntu to linux for sdk install
- redirect duplication to linux sdk install guide

Signed-off-by: Aymeric Aillet <aymeric.aillet@iot.bzh>

kartben reviewed (changes requested):
> Harmonize the sdk install part of the "install linux host dependencies" with the common sdk installation guide under develop/toolchains/zephyr_sdk.html.

"Harmonize" as-in "copy-paste" (not blaming you at all! you just followed what was already there) :| Not sure why I never realized we had duplicated content before.

The "Install the Zephyr Software Development Kit (SDK)" section should really just send the reader to https://docs.zephyrproject.org/latest/develop/toolchains/zephyr_sdk.html#zephyr-sdk-installation -- no ?

aaillet commented:
@kartben 
> > Harmonize the sdk install part of the "install linux host dependencies" with the common sdk installation guide under develop/toolchains/zephyr_sdk.html.
> 
> "Harmonize" as-in "copy-paste" (not blaming you at all! you just followed what was already there) :| Not sure why I never realized we had duplicated content before.
> 
Yep, exactly :smile:
I figured that if we're going to have duplicate text in the documentation, it might as well be exactly the same, to simplify maintenance.
> The "Install the Zephyr Software Development Kit (SDK)" section should really just send the reader to https://docs.zephyrproject.org/latest/develop/toolchains/zephyr_sdk.html#zephyr-sdk-installation -- no ?

It would indeed solve the lack of information and simplify maintenance, but do we want links between docs in this kind of guide? This could spoil the user experience.

I say this because, to simplify maintenance even further, we could also link to this page in the main “getting started” section when talking about SDK installation, but I don't think that's advisable.

aaillet commented:
So, from your POV @kartben, should I move sdk installation guide for all operating systems (Ubuntu,macOS,Windows,Fedora,Clear Linux,Arch Linux) to https://docs.zephyrproject.org/latest/develop/toolchains/zephyr_sdk.html#zephyr-sdk-installation and just put a link to it here https://docs.zephyrproject.org/latest/develop/getting_started/installation_linux.html#install-the-zephyr-software-development-kit-sdk ?

We may keep the current "base" sdk installation guide in the getting started guide or make it a link too later

Thanks for your answers

aaillet commented:
Hmm, forgot this, I'll check current status since 4.2 release and edit my changes :+1:

aaillet committed:
doc: update SDK supported architectures

Architecture list comes directly from SDK readme
This list is accurate since SDK 0.17.1

Signed-off-by: Aymeric Aillet <aymeric.aillet@iot.bzh>

aaillet commented:
Just pushed a new version:
- rename common linux sdk install guide from ubuntu to linux
- linux setup sdk installation guide is now sending reader to common linux sdk install guide

Also updated SDK supported architectures list in doc
- should it be in another PR ?
- I saw https://github.com/zephyrproject-rtos/zephyr/pull/90414 removing mention to nios II from kernel but current SDK readme is stil mentionning it, @kartben can you point me to what we really want ? thanks

kartben reviewed (changes requested):
Thanks for your patience 🫣

kartben commented:
@aaillet please rebase on main for your CI issue

aaillet commented on doc/develop/toolchains/zephyr_sdk.rst:
```diff
@@ -17,8 +17,12 @@ The Zephyr SDK supports the following target architectures:
 * ARM (32-bit and 64-bit; ARMv6, ARMv7, ARMv8; A/R/M Profiles)
+* Microblaze (32-bit)
 * MIPS (32-bit and 64-bit)
+* Nios II
```
Perfect :ok_hand:
